File: lib/core/reader/epub/epub_reader_engine.dart
import '../../model/book_format.dart';
import '../reader_document.dart';
import '../reader_engine.dart';
import 'epub_archive_parser.dart';

class EpubReaderEngine implements ReaderEngine {
  final _parser = EpubArchiveParser();

  @override
  BookFormat get format => BookFormat.epub;

  @override
  Future<ReaderDocument> open({
    required String filePath,
    String? initialLocator,
  }) async {
    final parsed = _parser.parseCombinedHtml(filePath);
    return EpubDocument(
      filePath: filePath,
      initialLocator: initialLocator,
      htmlContent: parsed.htmlContent,
      chapterCount: parsed.chapterCount,
    );
  }
}

File: core/reader/src/main/kotlin/com/readler/core/reader/EpubReaderEngine.kt
package com.readler.core.reader

import com.readler.core.model.BookFormat
import com.readler.core.reader.epub.EpubArchiveParser
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class EpubReaderEngine : ReaderEngine {
    override val format: BookFormat = BookFormat.EPUB
    private val parser = EpubArchiveParser()

    override suspend fun open(source: ReaderSource): ReaderDocument = withContext(Dispatchers.IO) {
        val parsed = parser.parseCombinedHtml(source.filePath)
        ReaderDocument.Epub(
            filePath = source.filePath,
            initialLocator = source.initialLocator,
            htmlContent = parsed.htmlContent,
            chapterCount = parsed.chapterCount
        )
    }
}
